src/test/java/com/bravo/user/service/PaymentServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package com.bravo.user.service;
+
+import com.bravo.user.App;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.PaymentRepository;
+import com.bravo.user.dao.repository.UserRepository;
+import com.bravo.user.exception.DataNotFoundException;
+import com.bravo.user.model.dto.PaymentDto;
+import com.bravo.user.util.TestUtil;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration(classes = {App.class})
+@ExtendWith(SpringExtension.class)
+@SpringBootTest()
+class PaymentServiceTest {
+
+    private final PaymentRepository paymentRepository = mock(PaymentRepository.class);
+    private final UserRepository userRepository = mock(UserRepository.class);
+    private final ResourceMapper resourceMapper = mock(ResourceMapper.class);
+
+    private final PaymentService underTest = new PaymentService(
+            paymentRepository,
+            userRepository,
+            resourceMapper
+    );
+
+    @Test
+    void retrieveByUserId() {
+        final String userId = "user-id";
+
+        when(userRepository.findById(userId)).thenReturn(Optional.of(TestUtil.getUserData()));
+        when(paymentRepository.findAllByUserId(userId)).thenReturn(TestUtil.getPayments());
```

Comment:
Is this better than a looser matcher and a `verify`?  Do you know the difference? 

```
when(userRepository.findById(anyString())).thenReturn(Optional.of(TestUtil.getUserData()));
...
verify(userRepository).findById(userId);
```

Reply:
In my opinion that depends, because if you have a more complex method then you have to mock every call or access to their properties, but if you send a complete object (not a mock) then your code will not throw a NPE, and instead of having many "when then" I think it is a bit cleaner to have real objects.
And about the verify, I forgot to add that, I tend to always add it.